src/plugins/controls/public/control_group/actions/edit_control_flyout.tsx

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { isEqual } from 'lodash';
+import React, { useState } from 'react';
+
+import { EmbeddableFactoryNotFoundError } from '@kbn/embeddable-plugin/public';
+
+import { DataControlInput, ControlEmbeddable, IEditableControlFactory } from '../../types';
+import { pluginServices } from '../../services';
+import { ControlGroupStrings } from '../control_group_strings';
+import { useControlGroupContainerContext } from '../control_group_renderer';
+import { ControlEditor } from '../editor/control_editor';
+
+export const EditControlFlyout = ({
+  embeddable,
+  closeFlyout,
+  removeControl,
+}: {
+  embeddable: ControlEmbeddable<DataControlInput>;
+  closeFlyout: () => void;
+  removeControl: () => void;
+}) => {
+  // Controls Services Context
+  const {
+    overlays: { openConfirm },
+    controls: { getControlFactory },
+  } = pluginServices.getServices();
+  // Redux embeddable container Context
+  const reduxContext = useControlGroupContainerContext();
+  const {
+    embeddableInstance: controlGroup,
+    actions: { setControlWidth, setControlGrow },
+    useEmbeddableSelector,
+    useEmbeddableDispatch,
+  } = reduxContext;
+  const dispatch = useEmbeddableDispatch();
+
+  // current state
+  const panels = useEmbeddableSelector((state) => state.explicitInput.panels);
+  const panel = panels[embeddable.id];
+
+  const [currentGrow, setCurrentGrow] = useState(panel.grow);
+  const [currentWidth, setCurrentWidth] = useState(panel.width);
+  const [inputToReturn, setInputToReturn] = useState<Partial<DataControlInput>>({});
+
+  const onCancel = () => {
+    if (
+      isEqual(panel.explicitInput, {
```

Comment:
As a follow-up in some other PR maybe we should look into fixing the bug where the confirm model is shown if you create a new control then immediately cancel.